src/main/java/step4/dto/RaceRecord.java

```diff
@@ -0,0 +1,34 @@
+package step4.dto;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class RaceRecord {
+
+    private final List<Car> carGroup;
+
+    public RaceRecord(List<Car> carGroup) {
+        this.carGroup = new ArrayList<>(carGroup);
+    }
+
+    public List<Car> getRaceRecord() {
+        return Collections.unmodifiableList(carGroup);
+    }
+
+    public List<CarName> findWinners() {
+        return carGroup.stream()
+                .filter(car -> car.getPosition().equals(findMaxPosition()))
+                .map(Car::getName)
+                .collect(Collectors.toList());
```

Comment:
매번 findMaxPosition으로 최대 이동거리를 새로 계산하는게 맞는 걸까요? 
최초 한 번만 계산해주면 되지 않을까요? 그리고 getter로 값을 꺼낸 다음 연산에 사용하는건 좋지 않습니다 getter는 정말 데이터의 계층이동 혹은 출력을 위해서만 사용해주세요!

```suggestion
        Position maxPosition = findMaxPosition();
        
        return carGroup.stream()
                .filter(car -> car.equalPosition(maxPosition))
                .map(Car::getName)
                .collect(Collectors.toList());
```

이런식으로도 만들수 있겠죠.